twenty-first/src/math/bfield_codec.rs

```diff
@@ -648,6 +690,116 @@ mod tests {
+        let encoded_of_trailing_zeros_poly = poly_w_trailing_zeros.encode();
+
+        assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn trailing_zeros_have_no_effect_on_encoding_poly_bfe_pbt(
+        mut polynomial: Polynomial<BFieldElement>,
+        #[strategy(0usize..30)] num_leading_zeros: usize,
+    ) {
+        let encoded = polynomial.encode();
+        polynomial
+            .coefficients
+            .extend(vec![BFieldElement::zero(); num_leading_zeros]);
+        let encoded_of_trailing_zeros_poly = polynomial.encode();
+        prop_assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn trailing_zeros_have_no_effect_on_encoding_poly_xfe_pbt(
+        mut polynomial: Polynomial<XFieldElement>,
+        #[strategy(0usize..30)] num_leading_zeros: usize,
+    ) {
+        let encoded = polynomial.encode();
+        polynomial
+            .coefficients
+            .extend(vec![XFieldElement::zero(); num_leading_zeros]);
+        let encoded_of_trailing_zeros_poly = polynomial.encode();
+        prop_assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn disallow_trailing_zeros_in_poly_encoding_bfe(
+        mut polynomial: Polynomial<BFieldElement>,
+        #[filter(!#leading_coefficient.is_zero())] leading_coefficient: BFieldElement,
+        #[strategy(1usize..30)] num_leading_zeros: usize,
+    ) {
+        polynomial.coefficients.push(leading_coefficient);
+        let vec_encoding = polynomial.coefficients.encode();
+        let poly_encoding = polynomial.encode();
+        prop_assert_eq!(
+            vec_encoding,
+            poly_encoding,
+            "This test expects equivalence of Vec and Polynomial encoding"
+        );
+
+        polynomial
+            .coefficients
+            .extend(vec![bfe!(0); num_leading_zeros]);
+        let vec_encoding_with_trailing_zeros = polynomial.coefficients.encode();
+        let decoding_result =
+            Polynomial::<BFieldElement>::decode(&vec_encoding_with_trailing_zeros);
+        assert!(
+            matches!(
+                decoding_result.unwrap_err(),
+                BFieldCodecError::TrailingZerosInPolynomialCoefficients
+            ),
+            "Decoding a polynomial from a list with trailing zeros must fail with expected error"
+        );
+    }
+
+    #[proptest]
+    fn disallow_trailing_zeros_in_poly_encoding_xfe(
+        mut polynomial: Polynomial<XFieldElement>,
+        #[filter(!#leading_coefficient.is_zero())] leading_coefficient: XFieldElement,
+        #[strategy(1usize..30)] num_leading_zeros: usize,
+    ) {
+        polynomial.coefficients.push(leading_coefficient);
+        let vec_encoding = polynomial.coefficients.encode();
+        let poly_encoding = polynomial.encode();
+        prop_assert_eq!(
+            vec_encoding,
+            poly_encoding,
+            "This test expects equivalence of Vec and Polynomial encoding"
```

Comment:
The `assert_eq!(a, b)` already expresses the expected equivalence. In my opinion, writing it again in English is duplication.

Reply:
This is a rather complicated test. This test is actually not about the equivalence of `Vec` and `Polynomial` encoding, rather that equivalence is a prerequisite for the next lines of the test that verify that given that equivalence, trailing zeros are not accepted.